lib/extensions/simple_form/components/html5_no_aria_required.rb

```diff
@@ -0,0 +1,15 @@
+module SimpleForm
+  module Components
+    module HTML5NoAriaRequired
```

Comment:
I was comparing this to [the original module after my proposed revisions](https://github.com/heartcombo/simple_form/pull/1823/files#diff-8af1501edf4226ef8c313c430baf2f3211449105f7f7a7ff4e4e16e942a7abd9) and the `@html5` instance variable and `html5?` method stood out to me. 

It's not very obvious, but these are important for how other inputs decide to assign certain attributes, like [`type="number"` for numeric inputs](https://github.com/heartcombo/simple_form/blob/8f77f598b32ed11b0d56c44481607a6e911bcecb/lib/simple_form/inputs/numeric_input.rb#L9-L12).

I don't know that we actually use a numeric input anywhere, but I think for the sake of preserving all of the original behaviors, we should incorporate everything related to `@html5` in the original module.